packages/backend/src/QueryEngine.ts

```diff
@@ -133,35 +178,88 @@ export class QueryEngine<DB> {
         query: TQuery,
         opts?: {
             /**
-             * The name of the database schema to execute
-             * your SynthQL query against
+             * The name of the database schema to
+             * execute your SynthQL query against
              *
              * e.g `public`
              */
             schema?: string;
         },
     ): Promise<QueryResult<DB, TQuery>> {
-        return await collectLast(
-            generateLast(
-                execute<DB, TQuery>(query, {
-                    executors: this.executors,
-                    defaultSchema: opts?.schema ?? this.schema,
-                    prependSql: this.prependSql,
-                }),
-            ),
+        return collectLast(
+            this.execute(query, {
+                schema: opts?.schema ?? this.schema,
+                returnLastOnly: true,
+            }),
         );
     }
 
-    compile<T>(query: T extends Query<DB, infer TTable> ? T : never): {
-        sql: string;
-        params: any[];
-    } {
-        const { sqlBuilder } = composeQuery({
-            defaultSchema: this.schema,
-            query,
+    // TODO: fix generic types for input and return types
+    // Currently returning `AsyncGenerator<never, any, unknown>`
```

Comment:
I don't think there is a way to fix this as the queryId and params have no knowledge of which table they come from. This will always return a QueryResult<AnyQuery>